python/ray/serve/scripts.py

```diff
@@ -454,6 +467,26 @@ def run(
             visualizer = GraphVisualizer()
             visualizer.visualize_with_gradio(handle)
         else:
+            if reload and not blocking:
+                raise click.ClickException(
+                    "The --non-blocking option conflicts with the --reload option."
+                )
+            if reload:
+                if working_dir:
+                    watch_dir = working_dir
+                else:
+                    watch_dir = "./"
```

Comment:
```suggestion
                else:
                    watch_dir = app_dir
```
`app_dir` defaults to `"."` and is the directory that the import_path is imported from